### ISSUE TITLE
Is there a way to put some metadata when saving the artcraft?

### ISSUE THREAD

**Issue by happyshows** [pbiecek/archivist#294]
Hi,

Currently, I am saving 'title' (character) as metadata together with a diagram (plotly object) into my local Repo. I used a list to wrap these two elements and save as one artcraft.
However, when it comes to getting metadata out, it seems that I will have to loop aread all artcraft in the Repo and then understand which I need to load.

Imaging if the repo has 1000 diagram and user only wants to select one based on the 'title' info. The performance would be really bad.
It seems to be really inefficient, is there a better way to do this?

**Comment by pbiecek** [comment]
Definitely,
you can use the 'userTags' argument of saveToLocalRepo() function to add any number of tags.
Then you can use the asearch() function to find/get artifacts with selected tags.
As `userTags` you can add a title, URL or other hooks

**Comment by happyshows** [comment]
cool thanks.

**Comment by happyshows** [comment]
Sorry need to reopen this.

Which function will give me all the relevant tags (and only the useful ones)

for example, I called splitTagsLocal()
                          artifact         tagKey                         tagValue         createdDate
1 9c55ee33f39269f329a61e2095c69438         format                              rda 2016-09-13 17:19:37
2 9c55ee33f39269f329a61e2095c69438           name                             adap 2016-09-13 17:19:37
3 9c55ee33f39269f329a61e2095c69438          class                             adap 2016-09-13 17:19:37
4 9c55ee33f39269f329a61e2095c69438           date              2016-09-13 17:19:37 2016-09-13 17:19:37
5 9c55ee33f39269f329a61e2095c69438 Packout_1                  Packout_1 2016-09-13 17:19:37
6 9edb8c4d783fc14e42399f23c5b58a69         format                              rda 2016-09-13 17:19:37
7 9c55ee33f39269f329a61e2095c69438   session_info 9edb8c4d783fc14e42399f23c5b58a69 2016-09-13 17:19:37

Packout_1 is the metadata I saved.

Will 
tags_df <- splitTagsLocal() %>% filter(tagKey == tagValue)
unique(tags_df$tagValue) give me an 'Index of Tags' which I can use in asearch()?
